connect/cli/plugins/project/extension/validations.py

```diff
@@ -234,64 +269,173 @@ def validate_schedulables(config, project_dir, context):
+                    ),
+                )
+                continue
+
+            if variable["name"] in names:
+                messages.append(
+                    ValidationItem(
+                        'ERROR',
+                        f'Duplicate variable name: the variable with name *{variable["name"]}* '
+                        'has already been declared.',
+                        **get_code_context(extension_class, '@variables'),
+                    ),
+                )
+
+            names.append(variable["name"])
+
+            if not variable_name_regex.match(variable['name']):
+                messages.append(
+                    ValidationItem(
+                        'ERROR',
+                        f'Invalid variable name: the value *{variable["name"]}* '
+                        f'does not match the pattern *{variable_name_pattern}*.',
+                        **get_code_context(extension_class, '@variables'),
+                    ),
+                )
+            if 'initial_value' in variable and not isinstance(variable['initial_value'], str):
+                messages.append(
+                    ValidationItem(
+                        'ERROR',
+                        f'Invalid *initial_value* attribute for variable *{variable["name"]}*: '
+                        f'must be a non-null string not *{type(variable["initial_value"])}*.',
+                        **get_code_context(extension_class, '@variables'),
+                    ),
+                )
+
+            if 'secure' in variable and not isinstance(variable['secure'], bool):
+                messages.append(
+                    ValidationItem(
+                        'ERROR',
+                        f'Invalid *secure* attribute for variable *{variable["name"]}*: '
+                        f'must be a boolean not *{type(variable["secure"])}*.',
+                        **get_code_context(extension_class, '@variables'),
+                    ),
+                )
 
-    for variable in variables:
-        if 'name' not in variable:
-            messages.append(
-                ValidationItem(
-                    'ERROR',
-                    'Invalid variable declaration: the *name* attribute is mandatory.',
-                    **get_code_context(extension_class, '@variables'),
-                ),
-            )
-            continue
+    return ValidationResult(messages, False, context)
 
-        if variable["name"] in names:
-            messages.append(
-                ValidationItem(
-                    'ERROR',
-                    f'Duplicate variable name: the variable with name *{variable["name"]}* '
-                    'has already been declared.',
-                    **get_code_context(extension_class, '@variables'),
-                ),
-            )
 
-        names.append(variable["name"])
+def validate_webapp_extension(config, project_dir, context):  # noqa: CCR001
```

Comment:
Remember that you should validate the static files defined in the extension.json exists in the static_root folder. If we define the "/static/index.html" in the extension.json we should validate that inside static_root we have the file index.html.